### ISSUE TITLE
Rework user feedback

### ISSUE THREAD

**Issue by r888888888** [r888888888/danbooru#3833]
Some changes to reduce the potential of feedback for harassment:

* [ ] Limit feedback to 300 characters
* [ ] Rename negative feedback to suggestion
* [ ] Change red color to blue

**Comment by deusexcalamus** [comment]
I don't think this is needed, tbh. I look at the user feedback semi-regularly and I don't really see much in the way of things that could be classified as 'harassment'; I don't see how limiting feedback to 130 characters will eliminate/reduce any supposed 'harassment'.

I *really* don't think that feedback should be expired, good/bad records should be kept as a motivator for the user in question/other users to do better, they shouldn't just disappear after a while (this also goes with a previous change made so that mods/admins can't delete feedback left on their account).

Renaming negative feedback to 'suggestion', I...don't really have any feeling either way

Changing the color I don't really have a feeling about either.

**Comment by crs00** [comment]
None of these changes make any sense. They negate the whole point of having a feedback system.
Limiting feedbacks to 130 characters gives you less to work with than a twitter post.
Expiring old feedbacks after a year is utterly ridiculous. It removes the incentive to improve or continue to do well because any feedback will just disappear after a year.
Negative feedbacks aren’t suggestions. They’re “you messed this up and refuse to change for the better, so here’s a permanent stain on your record”.
Blue? What even...

**Comment by nonamethanks** [comment]
Limiting character size will render feedbacks useless - they are useful precisely because they can be phrased so that users know what they are doing wrong. You'll notice from the feedback page that only a percentage of bans fit under that limit - a lot of bans even go well over it because 130 characters is simply not enough to explain why the feedback is being issued. And negative feedbacks are supposed to be issued only after contacting the user, so if that's unsuccessful then trying to explain the feedback as well via messages is not going to work. Even very small messages like https://danbooru.donmai.us/user_feedbacks?search%5Buser_id%5D=102330 go past the 200 characters.

In the same way, expiring old feedbacks will basically make keeping track of troublesome users impossible or a real pain. The first example I could think of is https://danbooru.donmai.us/user_feedbacks?search%5Buser_id%5D=104086 but there's plenty more. Feedbacks help keeping track of whether troublemakers improve or insist in committing the same disruptive actions. Making them expire after such a short term would neuter them. 

And finally, regarding the renaming/recolor, as deus said, negative feedbacks are supposed to be... Well, negative. What's the point of issuing a negative feedback if it's not negative anymore? Since they're to be meant as last resort, they're already used sparingly and only if someone is a repeated offender and/or has refused to reply to DMails and to previous neutral feedbacks - at least that's how they've been used by the vast majority of users so far. Most newcomers are just gonna shrug it off and ignore "suggestions" until they get banned, because they're not going to see them as something they should actually keep mind of. Blue simply does not pack as much punch as red.

If someone disagrees with a negative feedback they've received, they can always raise that up with the staff - if it's not removed then there's no issue, and if it is then the fault is with whoever issued it, not with the system. I don't see why the whole site should suffer from the action of selected individuals.

**Comment by r888888888** [comment]
I often get complaints from users complaining about feedback. Comments that may read as innocuous to you or me often cause strong emotional reactions. It's not that surprising: the user can't respond to the feedback without resorting to messages sent to admins and mods pleading for it to be purged. Most negative messages are confrontational, accusatory, and public. The red color and the negative label don't help with this.

What's the goal of feedback? Is it to shame a user, or to teach them? 

I assume it's to teach them. If that's the case, the red color and "negative" label don't do anything. A wall of text isn't useful; you should send that in a message. 

If the goal is to shame, then we need a more rigorous process about mandating private messaging before resorting to feedback. A lot of critics just dump feedback without warning (in the past year, across 157 negative non-ban related feedback, only 27 bothered to send a message within a 2 month time frame).

I'm willing to walk back on the expiration, and increase the size limit to 300 chars, which if your name isn't Soundtracker should be adequate to convey your message.

**Comment by BrokenEagle** [comment]
Maybe keep it the way it is, but make adding negative feedback require additional steps.  Something you said made me think about the way it's done in professional organizations which is basically a graduated response.  Danbooru could do something similar.

Allow users to attach the DMails they've had with the user to the feedback that would only be visible to Moderator+ (but show the number of Dmails that are tied to that feedback to everyone). This would establish a paper trail as it were and would demonstrate an unwillingness of the user to moderate their actions and the necessity for the a higher level of action (neutral feedback).  If the behavior does not improve, then the response could be elevated and a negative feedback could be added.  Allow this to also be tied to any preceding neutral feedbacks and additional Dmails. Moderators could review this paper trail if needed and decide if enough warning had been given ahead of time to justify the issuing of the negative feedback.

Maybe do the same for bans as well, i.e. allow moderators to attach DMails they themselves have issued and/or negative feedback the user has received.

With the above, the whole process would become more transparent and would establish more justification for the moderation actions it performs.

**Comment by nonamethanks** [comment]
If the issue is with negatives being handed too freely, why not start with adding a notice when a neutral/negative is selected in the feedback form, reminding people that they should contact the user first? 

And again, from the negatives search it appears feedbacks are most often given to repeated offenders.  Look for example at the last 100 feedbacks: https://danbooru.donmai.us/user_feedbacks?commit=Search&search%5Bcategory%5D=negative&utf8=%E2%9C%93&limit=100

If we exclude bans, which are not related to this discussion, the remaining feedbacks all cite previous misbehavings as reasions for the negative. Most users featured in that page have had previous neutrals or negatives - The only quesionable one I could find is https://danbooru.donmai.us/user_feedbacks?search%5Buser_id%5D=396936 , and a notice would serve as a warning to not immediately jump to the gun for trigger happy users. And I want to point out how that feedback is well within the first proposed limit.

Honestly, if the problem is with the action of one user, again, why not punish that user instead of everyone else? Need I remind that the last time an action like this was taken, the ability to replace posts ended up being removed from all approvers instead of punishing the responsible users, and now there's a single user who takes care of them regularly, forcing a single point of failure on the site?

[b]Edit[/b]: what BrokenEagle said too. Just make issuing negatives take an additional step or something if a notice on the feedback page is deemed not enough. Neutering a system that works well because of individuals abusing it only hurts everyone else, and the infringers will just find new ways to mess with the system anyway.

**Comment by nonamethanks** [comment]
I hate to double-post, but since you mentioned Soundtracker, might I point out that Apollyon's ban to ceres in response was too well past the limits you're suggesting? It's more around ~500 characters, in fact.

**Comment by kittey** [comment]
> What's the goal of feedback? Is it to shame a user, or to teach them?

From what I’ve seen, neutral feedback is often used to teach users. “You’re doing X wrong. You should go read Y.” Making it a feedback entry shows other users that the offending user has been noticed and doesn’t need to be messaged by multiple users about it.

When negative feedback is given, it’s often – though not always – for repeat offenders, vandals or users stepping way out of line. I consider it a way to alert mods+ about problematic users that should be banned or demoted. Sadly, usually nothing happens or the messenger gets shot instead.

The suggestions to remove/rework flagging and feedbacks look like an attempt to transform Danbooru into a feel-good community because some users that don’t get their way complain about having their feelings hurt. Are we going to become Facebook? Only likes/upvotes allowed? Danbooru became what it is now because we roughhouse and anyone who annoys too many other users gets told off and possibly kicked out, not buttered up.

**Comment by ma0ko** [comment]
Why make a callout about Soundtracker?
Soundtracker's negative feedback on Apollyon's account comes with a big weight and may be stressful to Apollyon, but Apollyon is a user with a large amount of activity. This large amount of activity has seen a lot of criticism from different people, and he doesn't seem willing to CONSIDER any of it.
The idea of too much emotional weight being dumped on people over negative feedback is flawed. The things that Soundtracker criticised about Apollyon's activity comes from observations that were made over a long period of time.
Heavy feedback that is likely to cause emotional stress will NOT happen to those who don't have a huge activity and long term history of criticism from many parties. That is to say IF feedbacks are policed properly. Logically, Soundtracker's kind of feedback would only happen to someone like Apollyon. If an extent of analysis and demoralizing statements happened to someone with:

- more recent points of criticism
- less overall activity on the site

then it should obviously be considered uncalled for (and typically never happen in the first place).

**Comment by KuuderesShadow** [comment]
I signed up to GitHub just to post this.

Even with the increase to 300 characters, that still leaves feedbacks like the one I left on Wiliam over the limit. The only ways to make this shorter would be to either be less helpful (including potentially by making myself harder to understand through use of shorthand) or to be less polite. Taking it to the extreme I could have just said "bad tagging of uploads" - it would get across the point but do absolutely nothing to help the user improve, and likely make them feel far more down than the actual feedback did. Is that really what you want to encourage?

Something like nonamethanks's feedback on Sal.N would be more than double the proposed limit, and could be cut down, but by cutting out informative and relevant information. Yes, that's a negative, but it's an entirely justified one - and we know it is entirely justified in part because of the contents which would be the first to get cut out from the character limit.

Honestly, my first instinct with character limits if I needed to say more (and couldn't cut it back by being less helpful to the user) would be to separate out (as sensibly as possible) what I had to say over multiple feedbacks rather than just leaving one. Wouldn't this be closer to harrassment? Doubtless this would then get me in trouble, but this would be a direct result of your actions, and I'd have no qualms in pointing this out (in private, of course).

And how does it actually solve what you are aiming to rectify? Feedback in the mid to high hundreds of character counts (the feedback on AdminPageAbe has two more examples, with Mikaeri's being 541 characters) but there are too many to list across the site) are generally the most constructive feedback of all. It doesn't do a thing to get rid of short, useless negatives and indeed actively encourages users to do that instead of give constructive criticism.

That's not to say that a character limit is not justified - if nothing else to prevent people from making far too long feedbacks to be of any worth. But a limit aimed at combating this should be in the thousands of characters, not the hundreds.

Soundtracker's feedback on Apollyon (which I imagine is the main motivation for this, especially with you specifically naming him above) was over three _thousand_ characters, not over three _hundred_. 

Of course, a lot of the stuff in that feedback is stuff that needed saying, and much of it indeed had been said in negatives given to him before, except that he kept abusing his powers to delete it. Again, it would otherwise have been done over 4 (or more) separate negative feedbacks. Would that really have been any less "harrassment" of Apollyon than sticking it in a single one?

With regards the other two things:

Regarding renaming it to "suggestion" - isn't this more along the lines of what a neutral feedback is supposed to be? Negatives go out for things like people who ignore "suggestions", or who are getting banned or should be banned (like spambots and sockpuppet accounts).

I couldn't care less what colour it is. I don't see how that prevents harrassment either, though.
